ldso/dynlink.c

```diff
@@ -1933,32 +1937,29 @@ void __dls3(sgxlkl_app_config_t *app_config, void *tos)
 	reloc_all(app.next);
 	reloc_all(&app);
 
-	__init_utls(&app.tls);
+	// __init_utls(&app.tls);
 
 	update_tls_size();
-	void *initial_tls = calloc(libc.tls_size, 1);
-	if (!initial_tls) {
-		dprintf(2, "%s: Error getting %zu bytes thread-local storage: %m\n",
-			argv[0], libc.tls_size);
-		_exit(127);
-	}
-
-    struct lthread *lt = lthread_self();
-    lt->itls = initial_tls;
-    lt->itlssz = libc.tls_size;
-	if (__init_utp(__copy_utls(lt, lt->itls, lt->itlssz), 1) < 0) {
-		a_crash();
-	}
-//	else {
-//		size_t tmp_tls_size = libc.tls_size;
-//		pthread_t self = __pthread_self();
-//		/* Temporarily set the tls size to the full size of
-//		 * builtin_tls so that __copy_tls will use the same layout
-//		 * as it did for before. Then check, just to be safe. */
-//		libc.tls_size = sizeof builtin_tls;
-//		if (__copy_tls((void*)builtin_tls) != self) a_crash();
-//		libc.tls_size = tmp_tls_size;
-//	}
+	//if (libc.tls_size > sizeof builtin_tls || tls_align > MIN_TLS_ALIGN) {
```

Comment:
Why are we not doing this check? Is this because the initial thread is not providing a tls segment at all?

Reply:
If the app has no tls variables libc.tls_size comes out to be 264. The builtin_tls is 360. This causes the else block to be evaluated, which tries to reuse the thread pointer created in `__dls2b`. Which causes failure. 
An alternative is we can remove the pthread_self() check in the else block and reinitialize the thread pointer if we want to do this check.

Reply:
Yes exactly.